apps/dgst.c

```diff
@@ -403,14 +403,32 @@ int dgst_main(int argc, char **argv)
         md_name = EVP_MD_get0_name(md);
 
     if (xoflen > 0) {
-        if (!(EVP_MD_get_flags(md) & EVP_MD_FLAG_XOF)) {
+        if ((EVP_MD_get_flags(md) & EVP_MD_FLAG_XOF) == 0) {
             BIO_printf(bio_err, "Length can only be specified for XOF\n");
             goto end;
         }
+        /*
+         * Signing using XOF is not supported by any algorithms currently since
+         * each algorithm only calls EVP_DigestFinal_ex() in their sign_final
+         * and verify_final methods.
+         */
         if (sigkey != NULL) {
             BIO_printf(bio_err, "Signing key cannot be specified for XOF\n");
             goto end;
         }
+    } else {
+        /*
+         * Print a warning if the default security strength is less than 128 bits,
+         * but don't fail if the -xoflen was not specified.
+         */
+        if ((EVP_MD_get_flags(md) & EVP_MD_FLAG_XOF) != 0)
+            BIO_printf(bio_err,
+                       "Warning: The -xoflen option was not set.\n"
+                       "By default the xoflen is set to half of the maximum security strength\n"
```

Comment:
External providers might have different XOF algorithms with different default xoflen that respects the security strength of the algorithm. So IMO this is slightly incorrect.

Reply:
Updated the text